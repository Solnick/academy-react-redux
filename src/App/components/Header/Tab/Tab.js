import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './tab.module.scss';

const Tab = ({ text, to }) => {
  const isActive = destination => window.location.href.endsWith(destination);
  return (
    <Link to={to} className={`${styles.tab} ${isActive(to) ? styles.activeTab : ''}`}>
      <p>{text}</p>
    </Link>
  );
};

Tab.propTypes = {
  text: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
};

export default Tab;
