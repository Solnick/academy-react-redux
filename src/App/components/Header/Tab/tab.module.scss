@import '../../../style/variables';
@import '../../../style/breakpoints';

.tab {
  color: $font-color;
  text-decoration: none;
  text-transform: uppercase;
  line-height: 35px;
  font-weight: 500;
  &.activeTab, &:hover {
    transition-duration: 200ms;
    color: $primary-color;
    box-shadow: 0 1px 0 $primary-color, inset 0 -1px 0 $primary-color;
  }
}