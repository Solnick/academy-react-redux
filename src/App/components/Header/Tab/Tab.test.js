import React from 'react';
import { shallow } from 'enzyme';
import Tab from './Tab';

describe('Tab component', () => {
  it('should render correctly', () => {
    const props = {
      to: '/about-us',
      text: 'about us',
      isActive: true,
    };
    const tabInstance = shallow(<Tab {...props} />);
    expect(tabInstance).toMatchSnapshot();
  });
});
