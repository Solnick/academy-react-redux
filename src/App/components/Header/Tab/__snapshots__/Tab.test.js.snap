// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tab
    isActive={true}
    text="about us"
    to="/about-us"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <p>
        about us
      </p>,
      "className": "tab activeTab",
      "to": "/about-us",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "about us",
      },
      "ref": null,
      "rendered": "about us",
      "type": "p",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <p>
          about us
        </p>,
        "className": "tab activeTab",
        "to": "/about-us",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "about us",
        },
        "ref": null,
        "rendered": "about us",
        "type": "p",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
